transactor/src/main/scala/io/mediachain/transactor/StateMachine.scala

```diff
@@ -34,18 +34,21 @@ object StateMachine {
   sealed abstract class JournalEvent extends Serializable
   case class JournalCommitEvent(entry: JournalEntry) extends JournalEvent
   case class JournalBlockEvent(ref: Reference) extends JournalEvent
+  
+  class JournalState extends Serializable {
+    var seqno: BigInt = 0                                     // next entry index
+    var index: MMap[Reference, ChainReference] = new MHashMap // canonical -> chain mapp
```

Comment:
this goes by the pojo serialization.
The documentation in copycat says that catalyst supports java.util.Map at the top level, but not sure exactly of the details.